### DISCUSSION TITLE
plus–minus ± scala.compiletime.ops?

### DISCUSSION THREAD

**Discussion by 3xau1o** [scala/scala3#17491] [General Discussion]
The following type operation is valid
```scala
import scala.compiletime.ops.int.+

val a: 2 + 2 = 4
```
Is also valid to have union type operations
```scala
import scala.compiletime.ops.int.-

val b: (2 + 2) | (2 - 2) = 4
```
Can it be less verbose with a`±` math operator?

```scala
var c: 2 +- 2 = 4 // ok
c = 0 // ok
```

**Comment by szymon-rd** [comment]
It may be a better idea to introduce a type like this in your codebase:
```scala
type +-[A <: Int, B <: Int] = A + B | A - B
```
The cost of introducing new types like this in stdlib is possible confusion and growing complexity. Other types in `compiletime.ops` are impossible to implement using just type defs, this one is.
